lib/improver/weights.py

```diff
@@ -30,123 +30,259 @@
 # POSSIBILITY OF SUCH DAMAGE.
 """Module to create the weights used to Blend data."""
 
```

Comment:
There are lots of exceptions raised in these methods, which I like. However, it would be useful to have `Raises` sections in each methods docstrings for reference.